File: Practice/AllDomains/CoreCS/Algorithms/Strings/CaesarCipherEncryption.js
/*	author:@shivkrthakur  */
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    var n = parseInt(readLine());
    var input = readLine();
    var k = parseInt(readLine());
    var output = "";
    var a = "a".charCodeAt(0);
    var z = "z".charCodeAt(0);
    var A = "A".charCodeAt(0);
    var Z = "Z".charCodeAt(0);
    k = k % 26;
    for(var i = 0; i < input.length; i++){
        var ascii = input.charCodeAt(i);
        var char = input[i];

        if(ascii >= a && ascii <= z)
        {
            var sum = ascii + (k % z);
            if(sum > z) char = String.fromCharCode((a + sum - 1) % z);
            else char = String.fromCharCode(sum);
        }
        else if (ascii >= "A".charCodeAt(0) && ascii <= "Z".charCodeAt(0))
        {
            var sum = ascii + (k % Z);
            if(sum > Z) char = String.fromCharCode((A + sum - 1) % Z);
            else char = String.fromCharCode(sum);
        }
        output += char;
    }
    console.log(output);
}

File: Practice/AllDomains/CoreCS/Algorithms/Strings/CaesarCipherEncryption.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution 
{
    static void Main(String[] args) 
	{
        int n = Convert.ToInt32(Console.ReadLine());
        string s = Console.ReadLine();
        int k = Convert.ToInt32(Console.ReadLine());
        
        int [] input = s.ToCharArray().Select(x => (int)x).ToArray();
        int a = (int)'a', z = (int)'z', A = (int)'A', Z = (int)'Z';
        //Console.WriteLine($"s:{s} a:{a} z:{z} A:{A} Z:{Z}");
        k = k % 26;
        for(int i = 0; i < n; i++)
        {
           int c = input[i];
           int sum = 0;
           //Console.WriteLine($"{(char)c} c:{c} k:{k} c+k:{c+k} {(char)(c + k)}");
           if(c >= a && c <= z)
           {
              c = (c + k);
              if(c > z){ c = a - 1 + (c % z); }
           }
           if(c >= A && c <= Z)
           {
              c = (c + k);
              if(c > Z) { c = A - 1 + (c % Z); }
           }
           input[i] = c;
        }
        Console.WriteLine(string.Join("", input.Select(x => (char)x).ToArray()));
        
    }
}
